scrapy_poet/downloadermiddlewares.py

```diff
@@ -91,6 +93,27 @@ def process_request(
         self.crawler.stats.inc_value("scrapy_poet/dummy_response_count")
         return DummyResponse(url=request.url, request=request)
 
+    def _skip_dependency_creation(self, request: Request, spider: Spider) -> bool:
+        """See: https://github.com/scrapinghub/scrapy-poet/issues/48"""
+
+        # No need to skip if the callback doesn't default to the parse() method
+        if request.callback is not None:
+            return False
+
+        # If the Request.cb_kwargs possess all of the cb dependencies, then no
+        # warning message should be issued.
+        signature_iter = iter(inspect.signature(spider.parse).parameters)
+        next(signature_iter)  # skip the first arg: response
+        cb_param_names = set(signature_iter)
```

Comment:
We might want to compute cb_param_names once, because inspection is costly sometimes (on pypy?). But let's not do it now.